src/naming/Naming_AST.ml

```diff
@@ -231,6 +231,15 @@ let lookup_nonlocal_scope id scopes =
       let _ = error tok "no outerscope" in
       None
 
+let has_block_scope (lang : Lang.t) =
+  match lang with
+  | Ruby
+  | Python
+  | Php ->
+      false
+  | _js_ when Lang.is_js lang -> false
```

Comment:
`| lang -> Lang.is_js lang` ? This feels a bit like `if cond then true else false`.

Reply:
Hope it's OK if I keep it the way it is.

The function explicitly lists languages without block scope. To me, it was easier to read. I added comment blocks to visually help that everything except the `_` case is false.